url10fb6bd43e_guwere_P2PSearch_tgz-pJ8-MainJ8/src/Sender.java

```diff
@@ -0,0 +1,42 @@
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.SocketException;
+import java.util.concurrent.BlockingQueue;
+
+public class Sender implements Runnable {
+    private DatagramSocket socket;
+    DatagramPacket message;
+    BlockingQueue sendQueue;
+
+    public Sender(BlockingQueue sQueue){
+        try {
+            socket = new DatagramSocket();
+        } catch (SocketException e) {
+            System.err.println("Could not create socket in Sender");
+            e.printStackTrace();
+        }
+        sendQueue = sQueue;
+    }
+
+    public void run(){
+        while(true){
+            try {
+                DatagramPacket sendPacket = (DatagramPacket)sendQueue.take();
+                socket.send(sendPacket);
+                System.out.println("Sending message...");
+            } catch (IOException e) {
+                System.err.println("Error sending packet in Sender");
+                e.printStackTrace();
+            } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)